### DISCUSSION TITLE
cuda 11.4 is out.

### DISCUSSION THREAD

**Discussion by hwu36** [NVIDIA/cutlass#284] [General]
The best compiler so far for Ampere.  

Cutlass wgrad kernels are improved by 14% in geomean when evaluating the layers of resnet-50.  The max improvement is 37%.  No regression in any layer.

11.4 adds many new features in `ld`, `cp.async` ptx (https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#ptx-isa-version-7-4).  The coming cutlass 2.6 will support `prefetch_size` which can slightly improve the performance of many kernels.  If you cannot wait, you can just add them to https://github.com/NVIDIA/cutlass/blob/master/include/cutlass/arch/memory.h and https://github.com/NVIDIA/cutlass/blob/master/include/cutlass/arch/memory_sm80.h.  For example, `cp.async.ca.shared.global.L2::128B`.  Moreover, please let us know if you find the new cache eviction policy feature is helpful to your applications.  We can consider to support them in the future releases.

**Comment by FindDefinition** [comment]
How does ```prefetch_size``` work? assume we have 128x32 tile shape for A, 128bit access, half data, threads in a warp prefetch overlapped global memory to L2?
![image](https://user-images.githubusercontent.com/73811981/124099477-2670bb00-da90-11eb-864b-c5bdf0a3f8f6.png)

**Reply by hwu36** [reply]
kind of.  It is not for `thread 0`.  L2 is shared among all SMs, not just for one thread.

**Reply by FindDefinition** [reply]
how to understand ```prefetch_size=64B/128B/256B```? doc says it's just a performance hint , how does the value of prefetch size affect SASS? (may be cache control in SASS?)

**Reply by hwu36** [reply]
Just use 128B.  We tested all three and 128B is the best one.
